src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -59,39 +59,53 @@ partial class ReceiveComponent
             }
 
             receiveComponent.BindQueues(configuration.transportSeam.QueueBindings);
-
-            pipelineSettings.Register("TransportReceiveToPhysicalMessageProcessingConnector", b =>
+            
+            var hasOwningSatellite = false;
+            foreach (var satelliteDefinition in configuration.SatelliteDefinitions.OfType<OwningSatelliteDefinition>())
             {
-                var storage = hostingConfiguration.Container.HasComponent<IOutboxStorage>() ? b.Build<IOutboxStorage>() : new NoOpOutboxStorage();
-                return new TransportReceiveToPhysicalMessageConnector(storage);
-            }, "Allows to abort processing the message");
+                satelliteDefinition.Name = MainReceiverId;
+                satelliteDefinition.ReceiveAddress = configuration.LocalAddress;
+                satelliteDefinition.RuntimeSettings = configuration.PushRuntimeSettings;
 
-            pipelineSettings.Register("LoadHandlersConnector", b =>
-            {
-                var adapter = hostingConfiguration.Container.HasComponent<ISynchronizedStorageAdapter>() ? b.Build<ISynchronizedStorageAdapter>() : new NoOpSynchronizedStorageAdapter();
-                var syncStorage = hostingConfiguration.Container.HasComponent<ISynchronizedStorage>() ? b.Build<ISynchronizedStorage>() : new NoOpSynchronizedStorage();
+                hasOwningSatellite = true;
+            }
 
-                return new LoadHandlersConnector(b.Build<MessageHandlerRegistry>(), syncStorage, adapter);
-            }, "Gets all the handlers to invoke from the MessageHandler registry based on the message type.");
+            var externalHandlerRegistryUsed = hostingConfiguration.Container.HasComponent<MessageHandlerRegistry>();
+            var handlerDiagnostics = new Dictionary<string, List<string>>();
+            
+            if (!hasOwningSatellite)
```

Comment:
Technically the code runs without that if. I just added it for demo purposes to show that we don't necessarily have to register all the other stuff that is never needed